app/src/main/java/org/wikipedia/descriptions/DescriptionEditActivity.java

```diff
@@ -93,13 +95,20 @@ private void copyLink(@NonNull String url) {
 
     @Override
     public DescriptionEditFragment createFragment() {
+        invokeSource = (InvokeSource) getIntent().getSerializableExtra(INTENT_EXTRA_INVOKE_SOURCE);
+        if (invokeSource == InvokeSource.EDIT_FEED_TITLE_DESC) {
+            SuggestedEditsFunnel.get().getAddDescriptionStats().click();
+        } else if (invokeSource == InvokeSource.EDIT_FEED_TRANSLATE_TITLE_DESC) {
+            SuggestedEditsFunnel.get().getTranslateDescriptionStats().click();
```

Comment:
Would it be good to send `InvokeSource` into the `SuggestedEditsFunnel` class and let the method do the logic and action? It might be easier to manage if we add more `InvokeSources` in the future. (e.g. `EDIT_FEED_IMAGE_CAPTION`)

Reply:
+1

Reply:
Correct!